nomad/acl.go

```diff
@@ -156,7 +156,29 @@ func (s *Server) remoteIPFromRPCContext(ctx *RPCContext) (net.IP, error) {
 	return nil, structs.ErrPermissionDenied
 }
 
-func (s *Server) ResolveACL(aclToken *structs.ACLToken) (*acl.ACL, error) {
+func (s *Server) ResolveACL(args structs.RequestWithIdentity) (*acl.ACL, error) {
```

Comment:
Sorry it's not formatted properly and probably wordier than it needs to be, but I thought I'd at least leave a sample for you to grab.

```suggestion
// ResolveACL is an authentication wrapper which handles resolving both ACL tokens and Workload Identities. If both are provided the ACL token is preferred, but it is best for the RPC caller to only include the credentials for the identity they intend the operation to be performed with.
func (s *Server) ResolveACL(args structs.RequestWithIdentity) (*acl.ACL, error) {
```